amp/WEB-INF/src/org/digijava/module/aim/action/CompareActivityVersions.java

```diff
@@ -1,363 +1,391 @@
+        boolean iscurrentworkspacemanager = false;
+        
+        TeamMember currentMember = (TeamMember) request.getSession().getAttribute("currentMember");
+        AmpTeamMember ampCurrentMember = TeamMemberUtil.getAmpTeamMember(currentMember.getMemberId());
+        
+        if (ampCurrentMember.getAmpMemberRole().getTeamHead()) {
+            iscurrentworkspacemanager = true;
+        }
+        if (ampCurrentMember.getAmpTeam().getAccessType().equalsIgnoreCase(Constants.ACCESS_TYPE_MNGMT)) {
+            ispartofamanagetmentworkspace = true;
+        }
+        
+        // If the current user is part of the management workspace or is not the
+        // workspace manager of a workspace that's not management then hide.
+        vForm.setAdvancemode(!ispartofamanagetmentworkspace & iscurrentworkspacemanager);
+    }
+
+    public ActionForward viewDifferences(ActionMapping mapping, ActionForm form, HttpServletRequest request,
+            HttpServletResponse response) throws Exception {
+
+        CompareActivityVersionsForm vForm = (CompareActivityVersionsForm) form;
+        vForm.setOutputCollectionGrouped(ActivityVersionUtil.compareActivities(vForm.getActivityOneId()));
+
+        return mapping.findForward("forward");
+    }
+
+    public ActionForward compareAll(ActionMapping mapping, ActionForm form, HttpServletRequest request,
+                                    HttpServletResponse response) throws Exception {
+
+        CompareActivityVersionsForm vForm = (CompareActivityVersionsForm) form;
+        vForm.populateEffectiveFilters();
+        vForm.setActivityComparisonResultList(ActivityVersionUtil.
+                getOutputCollectionGrouped(vForm.getEffectiveSelectedUser(), vForm.getEffectiveSelectedTeam(),
+                        vForm.getEffectiveDateFrom(), vForm.getEffectiveDateTo()));
+
+        return mapping.findForward("forward");
+    }
+    
+    public ActionForward xlsExport(ActionMapping mapping, ActionForm form, HttpServletRequest request,
+            HttpServletResponse response) throws Exception {
+        response.setContentType("application/vnd.ms-excel");
+        response.setHeader("Content-disposition", "inline; filename=AuditLogger.xls");
+        CompareActivityVersionsForm vForm = (CompareActivityVersionsForm) form;
+        Site site = RequestUtils.getSite(request);
+        Locale navigationLanguage = RequestUtils.getNavigationLanguage(request);
+        Long siteId = site.getId();
+        String locale = navigationLanguage.getCode();
+        if (vForm.getActivityOneId() == 0) {
+            List<ActivityComparisonResult> outputCollection = vForm.getActivityComparisonResultList();
+            HSSFWorkbook wb = AuditExcelExporter.generateExcel(locale, siteId, outputCollection);
+            wb.write(response.getOutputStream());
+        } else {
+            Map<String, List<CompareOutput>> outputCollectionGrouped = vForm.getOutputCollectionGrouped();

```

Comment:
this code seems to be dead, noone is calling it.
It might be usefull to create a List<ActivityComparisonResult> outputCollection of 1 object to reuse the same method

Reply:
what do you mean? It is being called by single Activity